### PR TITLE
Rework tx sending workflow, tx consolidation, and tx display

### LINKED ISSUE #89: (Production preview) wallet crashes when brought from bg to fg
You can download an APK here to test this out: https://expo.dev/accounts/nop33/projects/mobile-wallet/builds/48e7075a-d986-41f6-8b69-7f092ad80c01

### PR DESCRIPTION
**Pull request by nop33**
I am opening this PR against the `refresh` branch which has an open PR: #88

1. It improves the way pending txs are handled. Instead of merging "dummy" data into the confirmed transactions array in the app state, it stores them separately. This makes the code much cleaner, imo.
2. It handles the UTXO consolidation corner case, when the address the user tries to send funds from has too many UTXOs. Fixes https://github.com/alephium/mobile-wallet/issues/74
3. It updates the display of transactions to look sleek af (and aligned with how they look in all other apps of ours).
4. It simplifies A LOT the process of restoring the navigation state
5. It fixes #89 

Here's a preview (in shity quality due to GH filesize limit)

https://user-images.githubusercontent.com/1579899/197814106-f629b16d-13f3-4841-b70b-b30c53e3f49a.mov

### CODE REVIEW

**Comment by mvaivre** [comment]
That's magnificent.

File: src/screens/SendScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+  const [amountString, setAmountString] = useState('')
+  const [amount, setAmount] = useState(BigInt(0))
+  const [fees, setFees] = useState<bigint>(BigInt(0))
+  const [unsignedTxId, setUnsignedTxId] = useState('')
+  const [unsignedTransaction, setUnsignedTransaction] = useState('')
+  const [gasAmount, setGasAmount] = useState('')
+  const [gasPriceString, setGasPriceString] = useState('')
+  const [gasPrice, setGasPrice] = useState<bigint>()
+  const [isLoading, setIsLoading] = useState(false)
+  const [isConsolidateUTXOsModalVisible, setIsConsolidateUTXOsModalVisible] = useState(false)
```
Comment (mvaivre) [inline src/screens/SendScreen.tsx:83]: That's a lot of states 🙃
I wonder if we should maybe start considering using something like https://react-hook-form.com/ which would greatly simplify local form state management and validation.

Comment (nop33) [reply src/screens/SendScreen.tsx:83]: I'll have a look into it!

File: src/utils/transactions.ts [hunk 2]
```diff
@@ -18,18 +18,21 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
-    } else {
-      if (existingTransactions[targetTxIndex].blockHash === '') {
-        existingTransactions.splice(targetTxIndex, 1, incomingTx)
-      }
-    }
-    return newTxs
-  }, [])
+  incomingTransactions.filter((newTx) => !existingTransactions.some((existingTx) => existingTx.hash === newTx.hash))
+
+export const getRemainingPendingTransactions = (
```
Comment (mvaivre) [inline src/utils/transactions.ts]: Same comment as before, `extractRemainingPendingTransactions` to avoid confusion with endpoint call?

Comment (nop33) [reply src/utils/transactions.ts]: haha, already done!

Comment (mvaivre) [reply src/utils/transactions.ts]: Ah you did it already :D

File: src/screens/SendScreen.tsx (hunk 1 above)
Comment (nop33) [reply src/screens/SendScreen.tsx:83]: Hm, looking at the [integration process of React Hook Form into React Native](https://echobind.com/post/react-hook-form-for-react-native) and the changes that are needed to our codebase, I am not so sure it's worth going for it for one component that has a lot of state. WDYT? I could try reducing the use of `useState` by grouping things together...

Comment (nop33) [reply src/screens/SendScreen.tsx:83]: Or maybe I am wrong, wait: https://react-hook-form.com/get-started#reactnative

File: src/screens/SendScreen.tsx [hunk 3]
```diff
@@ -68,113 +81,119 @@ const SendScreen = ({
...
-      setGasPrice(convertAlphToSet(gasPriceString))
-    } catch {
-      setGasPrice(undefined)
-    }
-  }, [gasPriceString])
+  const isFormValid = isEmpty(errors)
+  const totalAmount = amount + fees
+
+  const validateIsAddressValid = (value: string) => isAddressValid(value) || 'This address is not valid'
+  const validateIsNumericStringValid = (value: string) => isNumericStringValid(value) || 'We need a number here!'
```
Comment (mvaivre) [inline src/screens/SendScreen.tsx:116]: Should we start to extract generic validation method out of this component?

Also, let's maybe replace the error by something like `A number is expected.` :)

**Comment by mvaivre** [comment]
One tiny comment. Otherwise, the code looks great to me! Well done!!

File: src/screens/SendScreen.tsx (hunk 3 above)
Comment (nop33) [reply src/screens/SendScreen.tsx]: done!

File: src/navigation/RootStackNavigation.tsx [hunk 4]
```diff
@@ -84,26 +78,12 @@ const RootStackNavigation = () => {
...
-        // Means this is a regular login, not coming back from being inactive
-      } else if (isAuthenticated && isAppBackgroundedAcknowledged && !isResetFromBeingInactive) {
-        rootStackNavigationRef.resetRoot({ index: 0, routes: [{ name: 'InWalletScreen' }] })
-      }
-    },
+  useEffect(() => {
+    if (isAuthenticated) {
+      rootStackNavigationRef.resetRoot(lastNavigationState ?? { index: 0, routes: [{ name: 'InWalletScreen' }] })
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (mvaivre) [inline src/navigation/RootStackNavigation.tsx:85]: Is this needed?

**Review by mvaivre — APPROVED** [review]
👏

**Pull request merged**